client/app/views/popover_screens/guests.coffee

```diff
@@ -158,31 +183,79 @@ module.exports = class GuestPopoverScreen extends EventPopoverScreenView
+            if cozy?
+                guestBisCozy  = _.findWhere(guests, cozy: cozy)
+
+            # But a duplicate is not a duplicate under certain circumstances:
+            # * same email and `shareWithCozy` is true;
+            # * same cozy and `shareWithCozy` is false;
+            # In those cases we can add the new guest.
+            if (email? and (not guestBisEmail? or
+            guestBisEmail?.shareWithCozy)) or (cozy? and (not guestBisCozy? or
+            (not guestBisCozy?.shareWithCozy)))
                 newGuest =
-                    key           : random.randomString()
-                    status        : 'INVITATION-NOT-SENT'
-                    email         : email
-                    label         : email
-                    contactid     : contactID
-                    shareWithCozy : false
-
-                # If guest was "autocompleted" then contactID is not null and we
-                # can check if a cozy instance is linked to this contact
+                    key       : random.randomString()
+                    status    : 'INVITATION-NOT-SENT'
+                    contactid : contactID
+
+                # If guest was "autocompleted" then contactID is not null. We
+                # can fill additionnal information.
                 if contactID?
                     contact       = app.contacts.get contactID
-                    newGuest.cozy = contact.get 'cozy'
                     newGuest.name = contact.get 'name'
+                    if email?
+                        newGuest.email         = email
+                        newGuest.label         = email
+                        # By default if a guest has several cozy linked to him,
+                        # the first one is chosen.
+                        newGuest.cozy          =
+                            (contact.get 'cozy')?[0]?.value or null
+                        newGuest.shareWithCozy = false
+                    else if cozy?
+                        emails                 = (contact.get 'emails')
+                        # By default if a guest has several emails linked to
+                        # him, the first one is chosen.
+                        newGuest.email         = emails?[0]?.value or null
+                        newGuest.label         = cozy
+                        newGuest.cozy          = cozy
+                        newGuest.shareWithCozy = true
+                # Guest was manually entered
+                else
+                    if email?
+                        newGuest.label         = email
+                        newGuest.email         = email
+                        newGuest.shareWithCozy = false
+                        newGuest.cozy          = null
+                    else
+                        newGuest.label         = cozy
+                        newGuest.email         = null
+                        newGuest.shareWithCozy = true
+                        newGuest.cozy          = cozy
 
```

Comment:
Maybe a little bit too much if/then/else here.

What about
```coffeescript
if contactID?
    contact = app.contacts.get contactID
_.extend newGuest,
    name: (contact and contact.get 'name') or null # or email ?
    cozy: (email and contact and contact.get('cozy')?[0]?.value) or cozy or null
    email: email or (cozy and contact and contact.get('emails')?[0]?.value) or null
    label: email or cozy
    shareWithCozy: cozy?
 ```
I did not test it, but there is the idea :)